### PR TITLE
T17: String permutations

### PR DESCRIPTION
**Pull request by Dakshita-Minocha**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+         string input;
+         for (; ; ) {
+            for (; ; ) {
+               Console.Write ("Enter word, or enter [Q] to exit: ");
+               input = Console.ReadLine ().ToLower ();
+               if (input == "q") Environment.Exit (0);
+               if (input.Length > 1 && input.All (c => Char.IsLetter (c))) break;
+            }
+            perm.Clear ();
+            perm = Permutations (input.ToList (), perm, k, l);
```
Comment (v-priya) [inline Program.cs:16]: Why `ToList ()`? Can simply be an array as they are indexable too. No specific advantage of using a list here

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
+      /// <param name="l"> Index for swapping </param>
+      /// <returns> List of all distinct permutations </returns>
+      static List<string> Permutations (List<char> input, List<string> perm, int k, int l) {
+         int len = input.Count;
+         string str = string.Join ("", input);
+         perm.Add (str);
+         (input[l], input[k]) = (input[k++], input[l++]);
+         if (l == len) l = 0;
+         if (k == len) k = 0;
+         if (perm.Count < NumofPermutations (len)) Permutations (input, perm, k, l);
```
Comment (v-priya) [inline Program.cs:38]: This will compute the `NumofPermutations` as many times the loop iteration happens. Cache it.

**Review by v-priya — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 1 above)
Comment (Dakshita-Minocha) [reply Program.cs:16]: I'm changing the value of the input in the function. Arrays aren't mutable right?

Comment (v-priya) [reply Program.cs:16]: You cannot add elements to an array without resizing. So, size is immutable. But, the elements inside an array can be changed.

**Review by v-priya — APPROVED** [review]

**Pull request closed without merging**